tests-e2e/client_test.go

```diff
@@ -38,635 +26,47 @@ type Event struct {
-	deviceID = gjson.GetBytes(body, "device_id").Str
-	return userID, accessToken, deviceID
-}
-
-// GetCapbabilities queries the server's capabilities
-func (c *CSAPI) GetCapabilities(t *testing.T) []byte {
-	t.Helper()
-	res := c.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "capabilities"})
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		t.Fatalf("unable to read response body: %v", err)
-	}
-	return body
-}
-
-// GetDefaultRoomVersion returns the server's default room version
-func (c *CSAPI) GetDefaultRoomVersion(t *testing.T) gomatrixserverlib.RoomVersion {
-	t.Helper()
-	capabilities := c.GetCapabilities(t)
-	defaultVersion := gjson.GetBytes(capabilities, `capabilities.m\.room_versions.default`)
-	if !defaultVersion.Exists() {
-		// spec says use RoomV1 in this case
-		return gomatrixserverlib.RoomVersionV1
-	}
-
-	return gomatrixserverlib.RoomVersion(defaultVersion.Str)
-}
-
-// MustDo will do the HTTP request and fail the test if the response is not 2xx
-//
-// Deprecated: Prefer MustDoFunc. MustDo is the older format which doesn't allow for vargs
-// and will be removed in the future. MustDoFunc also logs HTTP response bodies on error.
-func (c *CSAPI) MustDo(t *testing.T, method string, paths []string, jsonBody interface{}) *http.Response {
-	t.Helper()
-	res := c.DoFunc(t, method, paths, WithJSONBody(t, jsonBody))
-	if res.StatusCode < 200 || res.StatusCode >= 300 {
-		defer res.Body.Close()
-		body, _ := ioutil.ReadAll(res.Body)
-		t.Fatalf("CSAPI.MustDo %s %s returned HTTP %d : %s", method, res.Request.URL.String(), res.StatusCode, string(body))
-	}
-	return res
-}
-
-// WithRawBody sets the HTTP request body to `body`
-func WithRawBody(body []byte) RequestOpt {
-	return func(req *http.Request) {
-		req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
-		// we need to manually set this because we don't set the body
-		// in http.NewRequest due to using functional options, and only in NewRequest
-		// does the stdlib set this for us.
-		req.ContentLength = int64(len(body))
-	}
-}
-
-// WithContentType sets the HTTP request Content-Type header to `cType`
-func WithContentType(cType string) RequestOpt {
-	return func(req *http.Request) {
-		req.Header.Set("Content-Type", cType)
-	}
-}
-
-// WithJSONBody sets the HTTP request body to the JSON serialised form of `obj`
-func WithJSONBody(t *testing.T, obj interface{}) RequestOpt {
-	return func(req *http.Request) {
-		t.Helper()
-		b, err := json.Marshal(obj)
-		if err != nil {
-			t.Fatalf("CSAPI.Do failed to marshal JSON body: %s", err)
-		}
-		WithRawBody(b)(req)
-	}
-}
-
-// WithQueries sets the query parameters on the request.
-// This function should not be used to set an "access_token" parameter for Matrix authentication.
-// Instead, set CSAPI.AccessToken.
-func WithQueries(q url.Values) RequestOpt {
-	return func(req *http.Request) {
-		req.URL.RawQuery = q.Encode()
-	}
+	*client.CSAPI
+	Localpart string
+	Domain    string
+	AvatarURL string
```

Comment:
Should this be folded into complement?

Reply:
Maybe. Unsure for now, so leaving it here.